GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexParser.cs

```diff
@@ -105,21 +153,48 @@ private FileSystemTaskResult AddToProjection(GitIndexEntry data)
                 return FileSystemTaskResult.Success;
             }
 
-            private FileSystemTaskResult AddToModifiedFiles(GitIndexEntry data)
+            /// <summary>
+            /// Adjusts the modifed paths and placeholders list for an index entry.
+            /// </summary>
+            /// <param name="gitIndexEntry">Index entry</param>
+            /// <param name="filePlaceholders">
+            /// Dictionary of file placeholders.  AddEntryToModifiedPathsAndRemoveFromPlaceholdersIfNeeded will
+            /// remove enties from filePlaceholders as they are found in the index.  After
+            /// AddEntryToModifiedPathsAndRemoveFromPlaceholdersIfNeeded is called for all entries in the index 
+            /// filePlaceholders will contain only those placeholders that are not in the index.
+            /// </param>
+            private FileSystemTaskResult AddEntryToModifiedPathsAndRemoveFromPlaceholdersIfNeeded(
+                GitIndexEntry gitIndexEntry,
+                Dictionary<string, PlaceholderListDatabase.PlaceholderData> filePlaceholders)
             {
-                if (!data.SkipWorktree)
+                gitIndexEntry.BackgroundTask_ParsePath();

```

Comment:
At this point, we know that we're only going to use background-task-specific parts of the `GitIndexEntry` class, so we could cast it to a subclass, e.g. `GitIndexEntry.ForBackgroundTask`. That would allow the implementation of that class to group together those related methods and member variables, and not have to use name prefixes to keep them organized.

Though this approach could potentially have its own drawbacks too so take it with a grain of salt. Just an idea that could help clean up the `GitIndexEntry` and make the intent a bit more clear.

Reply:
> That would allow the implementation of that class to group together those related methods and member variables, and not have to use name prefixes to keep them organized.

After discussing this option with @sanoursa offline we decided that because this option wouldn't come for free at runtime (downcasts must be checked) we're better off leaving this as-is for now and we can revisit the design if we end up needing to expand it further.